bundles/binding/org.openhab.binding.plugwise/src/main/java/org/openhab/binding/plugwise/internal/PlugwiseBinding.java

```diff
@@ -68,142 +74,170 @@ protected void removeBindingProvider(PlugwiseBindingProvider bindingProvider) {
         super.removeBindingProvider(bindingProvider);
     }
 
-    @SuppressWarnings("rawtypes")
     @Override
-    public void updated(Dictionary config) throws ConfigurationException {
+    public void updated(Dictionary<String, ?> config) throws ConfigurationException {
 
-        if (config != null) {
+        if (config == null) {
+            return;
+        }
 
-            // First of all make sure the Stick gets set up
-            Enumeration keys = config.keys();
-            while (keys.hasMoreElements()) {
+        validateKeyPatternsInConfig(config);
 
-                String key = (String) keys.nextElement();
+        stick = setupStick(config);
 
-                // the config-key enumeration contains additional keys that we
-                // don't want to process here ...
-                if ("service.pid".equals(key)) {
-                    continue;
-                }
+        if (stick != null) {
+            setupNonStickDevices(config);
+            setProperlyConfigured(true);
+        } else {
+            logger.error("Plugwise needs at least one Stick in order to operate");
+        }
 
-                Matcher matcher = EXTRACT_PLUGWISE_CONFIG_PATTERN.matcher(key);
-                if (!matcher.matches()) {
-                    logger.error("given plugwise-config-key '" + key
-                            + "' does not follow the expected pattern '<PlugwiseId>.<mac|port|interval>'");
-                    continue;
-                }
+    }
 
-                matcher.reset();
-                matcher.find();
+    private Stick setupStick(Dictionary<String, ?> config) {
 
-                String plugwiseID = matcher.group(1);
+        String port = (String) config.get("stick.port");
 
-                if (plugwiseID.equals("stick")) {
-                    if (stick == null) {
+        if (port == null) {
+            return null;
+        }
 
-                        String configKey = matcher.group(2);
-                        String value = (String) config.get(key);
+        Stick stick = new Stick(port, this);
+        logger.info("Plugwise added Stick connected to serial port {}", port);
```

Comment:
This might be better as a `logger.debug`, unless users really need to see this message every single time.